tslib/src/azure/storage-helper.ts

```diff
@@ -97,15 +95,14 @@ export interface IBlobStorageHelper {
 }
 
 export class AsyncBlobService extends az.BlobService implements IBlobStorageService {
-
   constructor () {
     super(config.AzureStorage.ConnectionString)
   }
 
-  deleteContainerAsync (container: string): Promise<void> {
-    return new Promise((resolve, reject) => {
-      this.deleteContainer(container, (error, response) => {
-        if (error) {
+  async deleteContainerAsync (container: string): Promise<void> {
+    return await new Promise((resolve, reject) => {
```

Comment:
Shouldn't it be up to the caller to do the await'ing here? Awaiting on the return value of an async function is an anti-pattern.

Reply:
that's what i thought, until i read [this](https://jakearchibald.com/2017/await-vs-return-vs-return-await/)

Reply:
resolved - rules modified to disallow `return await` unless in `try..catch` block